### ISSUE TITLE
`import/named` linting issue with several React imports

### ISSUE THREAD

**Issue by LorrisSaintGenez** [import-js/eslint-plugin-import#2225]
Hello there,

Over the weekend, some linting issues started to appear and I can't find a way to make the errors disappear. (we're using `@types/react ^17.0.0`)
```
import { FC, ReactNode } from 'react';
```
and then, using eslint:
```
 1:10  error  FC not found in 'react'         import/named
 1:14  error  ReactNode not found in 'react'  import/named
```

Here's our config:
```
    "eslint": "7.18.0",
    "eslint-config-algolia": "16.0.0",
    "eslint-config-prettier": "7.2.0",
    "eslint-plugin-eslint-comments": "3.2.0",
    "eslint-plugin-import": "2.22.1",
    "eslint-plugin-prettier": "3.3.1",
    "eslint-plugin-react": "7.22.0",
    "eslint-plugin-react-hooks": "4.2.0",
```

Any clue? Lmk if you need more information

**Comment by ljharb** [comment]
Those are types. Use `import type` for them. Also make sure you’re using the TS resolver, because actual react doesn’t have those exports.

**Comment by melenudo** [comment]
Hi,

I'm facing the same issue with Stencil components:

```
import {
  Component,
  ComponentInterface,
  Element,
  Prop,
  Method,
  h,
  Host,
  Event,
  EventEmitter,
} from '@stencil/core';
```

This import works in 0.3.4 version of **eslint-import-resolver-node** and **is a legal import in typescript**.

"import type" is not a solution for Component, Element, ... because they are decorators and you'll get typescript errors like this:

```
Component' cannot be used as a value because it was imported using 'import type'
```

**Comment by ljharb** [comment]
I’m confused - a decorator is a runtime value and should be imported with `import`.

**Comment by melenudo** [comment]
> I’m confused - a decorator is a runtime value and should be imported with `import`.

That's right but with the last version of **eslint-import-resolver-node** I've got the same error of this issue, with the 0.3.4 I don't.
It's true that my comment it's confusing. I wrote that in the sense of the proposed solution (but has no sense because they are not types).
Finally I think that the solution it's using **only** the TS resolver but, until now (at least with the 0.3.4 version), the node resolver has been working with that imports without problems.

**Comment by ljharb** [comment]
@melenudo your solution is to list the TS resolver *first*, before the node one, so that it takes precedence - but can still fall back to the standard node resolution.

**Comment by TRMW** [comment]
I am also experiencing @melenudo's issue with my Stencil-generated web component library. Adding the TS resolver seems to fix the decorator imports, but causes the generated [global component element types](https://stenciljs.com/docs/typed-components#typed-components) (tl;dr: if your web component is named `Foo`, it will generate an `HTMLFooElement` global type, like [this example](https://github.com/ionic-team/stencil-component-starter/blob/master/src/components.d.ts#L25)) leading them to no longer be found. @melenudo have you encountered this as well?

Also if either of you could give an example of the what the config corresponding to @ljharb's recommendation to "list the TS resolver first, before the node one" would look like, I'd appreciate it, as I'm not sure I'm doing it correctly. This what I have:

```
  settings: {
    'import/resolver': {
      typescript: {},
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
```

Thanks in advance for any help.

**Comment by ljharb** [comment]
@TRMW you're doing what i suggested - putting the `node` key after the `typescript` key.

If you're still experiencing it, then that suggestion isn't useful.

Anyone who can provide a repro repo would be most appreciated.

**Comment by melenudo** [comment]
@TRMW, the global component types are working for me. I'm fixing an existing private project and I can't share it.
Basically is using the config you show:

```
module.exports = {
  extends: [
    './rules/typescript',
    './rules/imports',
    './rules/stencil',
    './rules/jsx-a11y.js',
    './rules/prettier',
  ].map(require.resolve),
  rules: {},
  env: {
    es6: true,
    node: true,
    jest: true,
    browser: true,
  },
  settings: {
    'import/resolver': {
      'typescript': {},
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
};
```

./rules/imports
```
module.exports = {
  plugins: ['import'],
  extends: ['plugin:import/typescript'],
  rules: {
    'import/no-unresolved': 'error',
    'import/named': 'error',
    'import/namespace': 'error',
    'import/default': 'error',
    'import/export': 'error',
    'import/no-named-as-default': 'warn',
    'import/no-named-as-default-member': 'warn',
    'import/no-duplicates': 'warn',
  },
};
```
./rules/typescript
```
module.exports = {
  parser: '@typescript-eslint/parser',
  extends: [
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
  ],
  rules: {
    ...
  },
};
```
